### ISSUE TITLE
Question about ImageNet (ILSVRC) directory structure from Kaggle dataset

### ISSUE THREAD

**Issue by lif314** [wimmerth/anyup#10]
Hi, thank you for the great work and for releasing this codebase!

I'm trying to train your model using the ImageNet (ILSVRC) dataset that I downloaded from the official Kaggle competition:
👉 [https://www.kaggle.com/competitions/imagenet-object-localization-challenge/data](https://www.kaggle.com/competitions/imagenet-object-localization-challenge/data)

However, I’m encountering a path issue when initializing the `ImageDataset` class.
The error message is:

```
FileNotFoundError(2, 'No such file or directory')
full_key: train_dataset
```

My dataset directory structure looks like this:

```
imagenet/
├── ILSVRC/
│   ├── Annotations/
│   ├── Data/
│   │   ├── CLS-LOC/
│   │   │   ├── train/
│   │   │   ├── val/
│   │   │   └── test/
│   └── ImageSets/
├── LOC_train_solution.csv
├── LOC_val_solution.csv
├── LOC_synset_mapping.txt
└── LOC_sample_submission.csv
```

According to the Kaggle data,

* The **training set** (`train/`) contains subfolders (e.g., `n01440764`, `n02119789`, ...).
* The **validation set** (`val/`) contains all images in a single folder, and the class labels are provided separately in `LOC_val_solution.csv`.

Could you please confirm which directory structure your `ImageDataset` expects for ImageNet?

* Should I point `root` to `.../ILSVRC/Data/CLS-LOC/train` directly?
* Or is the dataloader designed to read labels from `LOC_val_solution.csv` automatically?

If the code assumes a pre-organized `val/` directory (i.e., one subfolder per synset), do you recommend re-arranging it manually using the provided CSV file?

Thanks a lot for your help!
